### PR TITLE
fix: kmeshctl authz status exits 0 even when every pod query fails

### LINKED ISSUE #1880: Kmeshctl authz status exits 0 even when every pod query fails
kmeshctl authz status <nonexistent-pod> exits 0 even though it logs a failure:

kmeshctl authz status nonexistent-pod
# level=error msg="failed to get authz status for pod nonexistent-pod: ... not found"
# POD  AUTHORIZATION STATUS
# exit code: 0

Traced it to ctl/authz/authz.go, NewStatusCmd(). The per-pod loop logs the error and does continue on failure, but nothing tracks that it happened - table prints with just a header, command exits normally.

Same root cause as #1871 - error logged but never propagated to exit code.

Fix ready: track a failure count, exit 1 only if every requested pod failed (partial success across multiple pods still exits 0). Opening a PR.

### PR DESCRIPTION
**Pull request by bhumikadangayach**
Fixes #1880 

kmeshctl authz status was exiting 0 even when every requested pod's status fetch failed. The per-pod loop in NewStatusCmd() logged the error and continued, but never tracked failures - the table just printed with a header and no rows, and the command exited normally.

Same root cause as #1871 (kmeshctl version).

Added a failure counter to the loop, and exit 1 only if every requested pod failed - partial success across multiple pods still exits 0.

Verified: kmeshctl authz status <nonexistent-pod> now exits 1 instead of 0.

### CODE REVIEW

File: ctl/authz/authz.go [hunk 1]
```diff
@@ -120,10 +120,12 @@ func NewStatusCmd() *cobra.Command {
 			statuses := make([]podStatus, 0, len(podNames))
 
 			// Collect the status for each pod.
+			failed := 0
 			for _, podName := range podNames {
 				status, err := fetchAuthzStatus(cli, podName)
 				if err != nil {
 					log.Errorf("failed to get authz status for pod %s: %v", podName, err)
```
[the remark this answers is not in the document]
Comment (bhumikadangayach) [reply ctl/authz/authz.go]: Good catch, fixed - now using failedCount == len(podNames), which also handles the empty pod list case correctly.

File: ctl/authz/authz.go [hunk 2]
```diff
@@ -138,6 +140,10 @@ func NewStatusCmd() *cobra.Command {
 			}
 			tw.Flush()
 			fmt.Print(buf.String())
+
+			if failed > 0 && len(statuses) == 0 {
+				os.Exit(1)
+			}
```
[the remark this answers is not in the document]
Comment (bhumikadangayach) [reply ctl/authz/authz.go]: Added unit tests in authz_test.go covering all-failed, partial-success, all-succeeded, and the empty pod list edge case.

**Comment by yashisrani** [comment]
@bhumikadangayach to avoid DCO issue , always include Signed-off-by: Author Name <authoremail@example.com> in every commit message. You can also do this automatically by using the -s flag (i.e., git commit -s).

and e2e tests are failing . please fix it ..

**Comment by bhumikadangayach** [comment]
Fixed both - commits are signed off now and all E2E tests pass on the latest push.
